DllImportGenerator/DllImportGenerator/Marshalling/MarshallingGenerator.cs

```diff
@@ -379,6 +379,14 @@ private static IMarshallingGenerator CreateArrayMarshaller(TypePositionInfo info
 
         private static IMarshallingGenerator CreateCustomNativeTypeMarshaller(TypePositionInfo info, StubCodeContext context, NativeMarshallingAttributeInfo marshalInfo)
         {
+            if (marshalInfo.ValuePropertyType is not null && !context.CanUseAdditionalTemporaryState)
```

Comment:
This is a great branch and the message is clear. I do have a request on `CanUseAdditionalTemporaryState`. I see in the `StubContext` it is marked `abstract` but it isn't obvious as to why it would be true or false. Can you add a bit of commentary in `StubContext` on why one would or wouldn't support that. Usage here is clear but I don't fully understand all the cases when we would chose to set this to be true or false.